lab2/parser.py

```diff
@@ -0,0 +1,189 @@
+from sly import Parser
+from scaner import MyLexer
+
+# nie wiem czy akceptujemy na przykład x += "cos", "costam" < "cos2" itd. (pewnie do zmiany)
+# czy możena np. return k += 5
+
+class MyParser(Parser):
+    tokens = MyLexer.tokens
+    
+    precedence = (
+        ('left', ADD, SUB),
+        ('left', MUL, DIV),
+        ('left', DOTADD, DOTSUB),
+        ('left', DOTMUL, DOTDIV),
+        ('nonassoc', LTE, GTE, EQ, NEQ, LT, GT)
+    )
+    
+    @_('statements stmt_braces',
+       'stmt_braces')
+    def statements(self, p):
+        return None
+    
+    @_('";"',
+       'if_stmt',
+       'while_stmt',
+       'for_stmt',
+       'assign_expr',
+       'print_stmt',
+       'BREAK ";"',
+       'CONTINUE ";"',
+       'RETURN expr ";"')
+    def stmt(self, p):
+        return None
+    
+    @_('IF "(" bool_expr ")" stmt_braces',
+       'IF "(" bool_expr ")" stmt_braces ELSE stmt_braces')
+    def if_stmt(self, p):
+        return None
+    
+    @_('WHILE "(" bool_expr ")" stmt_braces')
+    def while_stmt(self, p):
+        return None
+    
+    @_('FOR ID "=" id_int ":" id_int stmt_braces')
+    def for_stmt(self, p):
+        return None
+    
+    @_('ID',
+       'INTNUM')
+    def id_int(self, p):
+        return None
+    
+    @_('"{" statements "}"',
+       'stmt')
+    def stmt_braces(self, p):
+        return None
+    
+    @_('PRINT print_rek ";"')
+    def print_stmt(self, p):
+        return None
+    
+    @_('print_rek "," value',
+       'value')
+    def print_rek(self, p):
+        return None
+    
+    @_('INTNUM',
+       'FLOAT',
+       'ID',
+       'STRING')
+    def value(self, p):
+        return None
+    
+    @_('SUB ID',
+       'SUB INTNUM',
+       'SUB FLOAT')
+    def inv_num(self, p):
+        return None
+    
+    @_('value',
+       'inv_num',
+       'bin_expr',
+       'assign_expr',
+       'relation_expr',
+       'matrix_bin_expr',
+       'matrix_transpose',
+       'matrix_funcs',
+       'matrix_ref')
+    def expr(self, p):
+        return None
+    
+    @_('bin_expr',
+       'relation_expr')
+    def bool_expr(self, p):
+        return None
+    
+    @_('expr ADD expr',
+       'expr SUB expr',
+       'expr MUL expr',
+       'expr DIV expr')
```

Comment:
Z tego co mi się wydaje, tutaj chcemy wykonywać już konkretne operacje, tak jak w [tutorialu](https://sly.readthedocs.io/en/latest/sly.html), np:

```python
    @_('expr PLUS expr')
    def expr(self, p):
        return p.expr0 + p.expr1
```

Także nie wiem czy możemy je zbić w grupę (podobnie w  pozostałych przypadkach)

Reply:
Właśnie kojarzę że mówił żeby akcji nie podejmować jeszcze. Ale nie jestem pewny

Reply:
Bo on teraz i tak mówi czy jest syntaxerror czy nie

Reply:
> Właśnie kojarzę że mówił żeby akcji nie podejmować jeszcze. Ale nie jestem pewny

W każdym razie można zapytać kogoś czy wie:))

Reply:
Racja, można dopytać, bo całkiem sporo to ułatwi

Reply:
Podpytałem i z tego co się dowiedziałem możemy póki co powstrzymać się od implementacji konkretnych działań :)